packages/flutter/lib/src/services/keyboard_inserted_content.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// A class representing rich content (such as a PNG image) inserted via the
+/// system input method.
+///
+/// The following data is represented in this class:
+///  - MIME Type
+///  - Bytes
+///  - URI
+@immutable
+class KeyboardInsertedContent {
+  /// Creates an object to represent content that is inserted from the virtual
+  /// keyboard.
+  ///
+  /// All parameters can be null.
+  const KeyboardInsertedContent({this.mimeType, this.uri, this.data});
+
+  /// Converts JSON received from the Flutter Engine into the Dart class.
+  KeyboardInsertedContent.fromJson(Map<String, dynamic>? metadata):
+      mimeType = metadata != null && metadata.isNotEmpty ? metadata['mimeType'] as String? : null,
+      uri = metadata != null && metadata.isNotEmpty ? metadata['uri'] as String? : null,
+      data = metadata != null && metadata.isNotEmpty ? Uint8List.fromList(
+          List<int>.from(metadata['data'] as Iterable<dynamic>)) : null;
+
+  /// The mime type of the inserted content.
+  final String? mimeType;
```

Comment:
when will `content.mimeType` be null?

Reply:
It most likely won't ever be null, but just in case something goes awry from the engine side, I am keeping it nullable. Better safe than sorry :)

Reply:
If the underlying API promises that this is non-nullable I think it's better to keep the type non-nullable. I think that will help us catch bugs in the engine/platform instead of relying on developers to handle the potential bugs.

Reply:
I looked at the API and made relevant parameters non-nullable.